src/Raven.Client/Documents/Session/AsyncDocumentSession.ConditionalLoad.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Commands;
+using Sparrow.Json;
+
+namespace Raven.Client.Documents.Session
+{
+    public partial class AsyncDocumentSession
+    {
+        /// <inheritdoc />
+        public async Task<(T Entity, string ChangeVector)> ConditionalLoadAsync<T>(string id, string changeVector, CancellationToken token = default)
+        {
+            using (AsyncTaskHolder())
+            {
+                if (id == null)
+                    throw new ArgumentNullException(nameof(id));
+
+                if (Advanced.IsLoaded(id))
+                {
+                    var e = await LoadAsync<T>(id, token).ConfigureAwait(false);
+                    var cv = Advanced.GetChangeVectorFor(e);
+                    return (e, cv);
+                }
+
+                if (changeVector == null)
```

Comment:
I think we want to throw that here. Or just continue.
Thoughts?

Reply:
Right now we are returning that the item does not exists

Reply:
Currently we allow to check if the entity loaded into session with `changeVector=null`
I think if it's not loaded then we should throw an exception

Reply:
Why, that is why we have `IsLoaded` for, no?